aquifer/aquifer
# Environment API

This PR introduces an Environment API that does a couple things:
- Allows users to add and removing environments.
- Allows users to ping those environments and make sure they a) exist and b) are accessible.
- Improves init process a bit
- Adds a global `--env` or `-e` flag that allows users to specify a valid environment for the current aquifer instance.
- Assigns an `Aquifer.environment` property that is false if no env specified, and an instance of Environment if an env is specified.
## Steps to test
- [x] Create a new aquifer project.
- [x] cd to the root of the new project
- [x] Run `aquifer` and ensure you see a few new commands: `envs`, `env-add`, `env-remove`, and `ping`, and note that there is a new global flag, `--env`.
- [x] Run `aquifer envs` and ensure it tells you that you do not have any environments yet.
- [x] Run `aquifer env-add` and step through the questions. Specify connection details to an actual environment.
- [x] Run `aquifer env-list` and ensure it returns your env.
- [x] Run `aquifer ping {env_name}` and ensure it returns a connection error object if it can't connect, and a success message if it can.
- [x] Run `aquifer build --env {env_name}` and ensure the build completes normally (doesn't pay attention to env yet, just allows it to be specified and loads it by default)
- [x] Run `aquifer env-remove {env_name}` and ensure it removes the env you specified.

Upcoming steps include building the Run API class so that all code uses the Run API to execute commands, and the Run API uses the `Aquifer.environment` property to determine where the commands should be run.
